plugins/janus_videoroom.c

```diff
@@ -2208,6 +2189,68 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
 	}
 }
 
+static void janus_videoroom_recorder_create(janus_videoroom_participant *participant, gboolean audio, gboolean video) {
```

Comment:
Both are called in a [longer critical section](https://github.com/andreasg123/janus-gateway/blob/7b7ab2e0fd3e7415dfd0372018ac61ad5be0195e/plugins/janus_videoroom.c#L2843) protected by `rec_mutex`. Currently, `rec_mutex` also protects `recording_active` and `recording_base` that are changed outside those functions.  Do mutexes properly deal with being locked a second time by the same thread (and then unlocked twice)?